protocol/x/vault/keeper/orders.go

```diff
@@ -344,6 +367,26 @@ func (k Keeper) GetVaultClobOrders(
 		orders[2*i+1] = constructOrder(clobtypes.Order_SIDE_BUY, i, orderIds[2*i+1])
 	}
 
+	if vaultParams.Status == types.VaultStatus_VAULT_STATUS_CLOSE_ONLY {
+		// In close-only mode with non-zero leverage.
+		reduceOnlyOrders := make([]*clobtypes.Order, quotingParams.Layers)
+		reduceOnlyMaxOrderSize := k.GetVaultInventoryInPerpetual(ctx, vaultId, perpetual.Params.Id)
+		stepSize := lib.BigU(clobPair.StepBaseQuantums)
+		reduceOnlyMaxOrderSize.Quo(reduceOnlyMaxOrderSize, stepSize)
+		reduceOnlyMaxOrderSize.Mul(reduceOnlyMaxOrderSize, stepSize)
+		// If vault is long, only need sell orders (even indices).
```

Comment:
non-blocking: I wonder if this code could be built slightly more future-proof by setting some `closeOnlySide` based on long/short, then filtering out the orders from the generated `orders` rather than using even/odd indices.